src/Sarif/Map/JsonMapBuilder.cs

```diff
@@ -15,58 +15,46 @@ namespace Microsoft.CodeAnalysis.Sarif.Map
     ///  JsonMapBuilder constructs a JsonMap for input Json files. It is passed
     ///  a size percentage target which controls how much detail it includes.
     /// </summary>
-    public class JsonMapBuilder
+    public static class JsonMapBuilder
     {
-        private const long NodeSizeEstimateBytes = 90;         // Bytes for one Node with containing property but not including children
```

Comment:
>private [](start = 8, length = 7)

I suggest a design that splits the difference between the previous design ("the one and only setting (maxFileSizePercentage) is a property of the builder class") and your new proposed design ("the settings are in a separate class, and the new Settings class performs calculations and has variable state").

I think a settings class is fine, but IMO a settings class should contain only the values the caller asks for. If the builder needs to adjust the settings, it's the builder that should perform the calculation, and if there's a need for the caller to find out what changes the builder made, then the adjusted properties should be properties of the builder.

Reply:
Good feedback. I struggled on the proper design; I've now separated two classes: JsonMapSettings (user preferences) and internal JsonMapRunSettings (user + calculated). The conversion happens in JsonMapBuilder.Build, when the file size is known.

This design now leaves a simple public static API for someone who wants a map (they don't need to construct a JsonMapBuilder instance for no clear reason), is thread safe, and safely allows reusing JsonMapSettings when simultaneously building multiple files (not that I'd expect that use commonly).